lib/api/multiObjectDelete.js

```diff
@@ -166,6 +169,60 @@ function _parseXml(xmlToParse, next) {
     });
 }
 
+/**
+ * processObjectVersion - process object version to be deleted
+ * @param {object} entry - entry from data model
+ * @param {string} bucketName - bucket name
+ * @param {function} next - callback to call with error or decoded version
+ * @return {undefined}
+ **/
+function processObjectVersion(entry, bucketName) {
+    let decodedVersionId;
+    if (entry.versionId) {
+        decodedVersionId = entry.versionId === 'null' ?
+            'null' : versionIdUtils.decode(entry.versionId);
+    }
+    if (decodedVersionId instanceof Error) {
+        monitoring.promMetrics('DELETE', bucketName, 404,
```

Comment:
updating the `promMetrics` does not make sense here : this is not associated to the purpose of this function (decoding version id?), but to the overal process... and so should be kept somewhere in `getObjMetadataAndDelete` (eg keep level of abstraction)

Reply:
See above comment, it's now in the last callback, in case of error, I removed this one